src/Connection/CallbackConnectionProvider.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace SMW\Connection;
+
+use Closure;
+
+/**
+ * @license GNU GPL v2+
+ * @since 3.0
+ *
+ * @author mwjames
+ */
+class CallbackConnectionProvider implements ConnectionProvider {
+
+	/**
+	 * @var Closure
+	 */
+	private $callback;
+
+	/**
+	 * @var mixed
+	 */
+	private $connection;
+
+	/**
+	 * @since 3.0
+	 *
+	 * @param Closure $callback
+	 */
+	public function __construct( Closure $callback ) {
```

Comment:
`callable` is less restrictive and has no downsides as far as I know

Reply:
Yes, but nothing is wrong with `Closure` and it doesn't impose a functional impairment.

Reply:
It does: you can only provide Closures like this. You can't pass in 'globalFunctionName', 'SomeClass::staticFunction' or $instanceOfClassThatIsCallable

Reply:
Right, yes that is the only valid functional argument I have seen in this thread that should be reconsidered to allow for something like [ $this, '...' ]. [0] requires PHP 5.4 which is fine given that min req. is PHP 5.5.

[0] http://php.net/manual/de/language.types.callable.php